plasmapy/simulation/particletracker.py

```diff
@@ -156,36 +182,51 @@ def boris_push(self, init=False):
         .. [1] C. K. Birdsall, A. B. Langdon, "Plasma Physics via Computer
                Simulation", 2004, p. 58-63
         """
-        dt = -self.dt / 2 if init else self.dt
-        b, e = self._interpolate_fields()
-
-        # add first half of electric impulse
-        vminus = self.v + self.eff_q * e / self.eff_m * dt * 0.5
-
-        # rotate to add magnetic field
-        t = -b * self.eff_q / self.eff_m * dt * 0.5
-        s = 2 * t / (1 + (t * t).sum(axis=1, keepdims=True))
-        vprime = vminus + np.cross(vminus.si.value, t) * u.m / u.s
-        vplus = vminus + np.cross(vprime.si.value, s) * u.m / u.s
-
-        # add second half of electric impulse
-        v_new = vplus + self.eff_q * e / self.eff_m * dt * 0.5
-
-        self.v = v_new
-        if not init:
-            self.x += self.v * dt
-
+        b = self._b()
+        e = self._e()
+        x = self._x
+        v = self._v
+        if init:
+            self._boris_push(self._x,
+                             self._v,
+                             b, e, -0.5 * self._hqmdt, -0.5*self._dt)
+            self._x -= self._v * -0.5 * self._dt
+        else:
+            self._boris_push(self._x,
+                             self._v,
+                             b, e, self._hqmdt, self._dt)
+
+    @staticmethod
+    @numba.njit(parallel=True)
+    # @profile
+    def _boris_push(x, v, b, e, hqmdt, dt):
+        for i in numba.prange(len(x)):
```

Comment:
this is thus parallelized over the number of particles and goes pretty darn fast!